week4/Login/src/components/SignUpPage.jsx

```diff
@@ -0,0 +1,239 @@
+import { useNavigate } from "react-router-dom";
+import styled from "styled-components";
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+const SignUpPage = () => {
+  const navigate = useNavigate();
+
+  const [id, setId] = useState("");
+  const [pw, setPw] = useState("");
+  const [confirmPw, setConfirmPw] = useState("");
+  const [name, setName] = useState("");
+
+  const [isButtonAble, setIsButtonAble] = useState(false);
+  const [isCheckBtnAble, setIsCheckBtnAble] = useState();
+
+  useEffect(() => {
+    const allowButtonClick = () => {
+      if (id && pw && confirmPw && name && pw === confirmPw && isCheckBtnAble) {
+        setIsButtonAble(true);
+      } else {
+        setIsButtonAble(false);
+      }
```

Comment:
```suggestion
      id && pw && confirmPw && name && pw === confirmPw && isCheckBtnAble ? setIsButtonAble(true) : setIsButtonAble(false)
```
요로케 삼항연산자 이용해서 정리해주는건 어때??

Reply:
완전 좋은 생각! 반영해서 리팩해볼게!